xonsh/commands_cache.py

```diff
@@ -73,7 +73,14 @@ def get_possible_names(name):
     @property
     def all_commands(self):
         paths = builtins.__xonsh_env__.get('PATH', [])
-        pathset = frozenset(x for x in paths if os.path.isdir(x))
+        def removeDups(p):
+            ret = list()
+            for e in p:
+                if e not in ret:
+                    ret.append(e)
+            return ret
+        path = removeDups(path)
+        pathset = tuple(x for x in paths if os.path.isdir(x))
```

Comment:
Not a blocker, but if this is no longer a set I would prefer if the variable name changed to reflect that.

Reply:
Yeah, I agree! Thanks for bringing it up.  maybe `valid_path` is a better name